lib/compat/wordpress-6.7/class-gutenberg-rest-templates-controller-6-7.php

```diff
@@ -0,0 +1,168 @@
+	 * @return string                            Original source of the template one of theme, plugin, site, or user.
+	 */
+	private static function get_wp_templates_original_source_field( $template_object ) {
+		if ( 'wp_template' === $template_object->type ) {
+			// Added by theme.
+			// Template originally provided by a theme, but customized by a user.
+			// Templates originally didn't have the 'origin' field so identify
+			// older customized templates by checking for no origin and a 'theme'
+			// or 'custom' source.
+			if ( $template_object->has_theme_file &&
+			( 'theme' === $template_object->origin || (
+				empty( $template_object->origin ) && in_array(
+					$template_object->source,
+					array(
+						'theme',
+						'custom',
+					),
+					true
+				) )
+			)
+			) {
+				return 'theme';
+			}
+
+			// Added by plugin.
+			if ( $template_object->has_theme_file && 'plugin' === $template_object->origin ) {
+				return 'plugin';
+			}
+
+			// Added by site.
+			// Template was created from scratch, but has no author. Author support
+			// was only added to templates in WordPress 5.9. Fallback to showing the
+			// site logo and title.
+			if ( empty( $template_object->has_theme_file ) && 'custom' === $template_object->source && empty( $template_object->author ) ) {
+				return 'site';
+			}
+		}
+
+		// Added by user.
+		return 'user';
+	}
+
+	/**
+	 * Returns a human readable text for the author of the template.
+	 *
+	 * @param WP_Block_Template $template_object Template instance.
+	 * @return string                            Human readable text for the author.
+	 */
+	// @core-merge: Nothing has changed in this function, the only reason to include it here is that it's a private function.
+	private static function get_wp_templates_author_text_field( $template_object ) {
+		$original_source = self::get_wp_templates_original_source_field( $template_object );
+		switch ( $original_source ) {
+			case 'theme':
+				$theme_name = wp_get_theme( $template_object->theme )->get( 'Name' );
+				return empty( $theme_name ) ? $template_object->theme : $theme_name;
+			case 'plugin':
+				if ( ! function_exists( 'get_plugins' ) ) {
+					require_once ABSPATH . 'wp-admin/includes/plugin.php';
+				}
+				$plugins = get_plugins();
+				// @core-merge: Prioritize plugin name instead of theme name for plugin-registered templates.
+				$plugin_name = isset( $template_object->plugin ) ? $template_object->plugin . '/' . $template_object->plugin : $template_object->theme;
+				$plugin      = $plugins[ plugin_basename( sanitize_text_field( $plugin_name . '.php' ) ) ];
+				return empty( $plugin['Name'] ) ? $template_object->theme : $plugin['Name'];
+				// @core-merge: End of changes to merge in core.
+			case 'site':
+				return get_bloginfo( 'name' );
+			case 'user':
+				$author = get_user_by( 'id', $template_object->author );
+				if ( ! $author ) {
+					return __( 'Unknown author' );
+				}
+				return $author->get( 'display_name' );
```

Comment:
Consider that the author may be deleted and their posts may be assigned incorrectly to the deleted users ID.

Reply:
I'm not sure to understand this point. In which cases can a post be assigned to a deleted user? This code was copy-and-paste from [what's currently shipped in WordPress](https://github.com/WordPress/wordpress-develop/blob/922949fb97334ec0bff551afb0bd8116ff8ade7c/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L868-L872), so if it needs to be fixed we might want to create a separate PR for this?

Reply:
I'd like some more details on that too.. Additionally, it is in core right now and could be handled there and not in this PR, unless it's a very small change.

Reply:
> This code was copy-and-paste from [what's currently shipped in WordPress](https://github.com/WordPress/wordpress-develop/blob/922949fb97334ec0bff551afb0bd8116ff8ade7c/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L868-L872),

Cool, let's leave it for now. I think I misread it when I reviewed it last month and didn't notice it handled the unknown author use case.